$MirOS: ports/print/cups/patches/patch-backend_serial_c,v 1.1.7.1 2005/03/18 15:51:00 tg Exp $
--- backend/serial.c.orig	Fri Apr 14 19:03:39 2006
+++ backend/serial.c	Tue May  9 17:48:47 2006
@@ -683,7 +683,7 @@ list_devices(void)
 
   for (i = 0; i < 100; i ++)
   {
-    sprintf(device, "/dev/ttyS%d", i);
+    snprintf(device, sizeof(device), "/dev/ttyS%d", i);
 
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
@@ -720,9 +720,10 @@ list_devices(void)
     }
   }
 
+# if (defined(__OpenBSD__))||(defined(__MirBSD__))||(defined(__ekkoBSD__))
   for (i = 0; i < 16; i ++)
   {
-    sprintf(device, "/dev/usb/ttyUSB%d", i);
+    snprintf(device, "/dev/ttyU%d", i);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
@@ -745,6 +746,18 @@ list_devices(void)
       }
     }
   }
+# else
+  for (i = 0; i < 16; i ++)
+  {
+    sprintf(device, "/dev/usb/ttyUSB%d", i);
+    if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
+    {
+      close(fd);
+      printf("serial serial:%s?baud=230400 \"Unknown\" \"USB Serial Port #%d\"\n",
+             device, i + 1);
+    }
+# endif
+
 #elif defined(__sgi)
   int		i, j, n;	/* Looping vars */
   char		device[255];	/* Device filename */
@@ -970,7 +983,7 @@ list_devices(void)
 
   for (i = 0; i < 32; i ++)
   {
-    sprintf(device, "/dev/ttyd%c", funky_hex[i]);
+    snprintf(device, sizeof(device), "/dev/ttyd%c", funky_hex[i]);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
@@ -986,7 +999,7 @@ list_devices(void)
   for (i = 0; i < 16; i ++) /* Should be up to 65536 boards... */
     for (j = 0; j < 32; j ++)
     {
-      sprintf(device, "/dev/ttyc%d%c", i, funky_hex[j]);
+      snprintf(device, sizeof(device), "/dev/ttyc%d%c", i, funky_hex[j]);
       if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
       {
 	close(fd);
@@ -1010,7 +1023,7 @@ list_devices(void)
   for (i = 0; i < 16; i ++) /* Should be up to 65536 boards... */
     for (j = 0; j < 32; j ++)
     {
-      sprintf(device, "/dev/ttyD%d%c", i, funky_hex[j]);
+      snprintf(device, sizeof(device), "/dev/ttyD%d%c", i, funky_hex[j]);
       if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
       {
 	close(fd);
@@ -1025,7 +1038,7 @@ list_devices(void)
 
   for (i = 0; i < 32; i ++)
   {
-    sprintf(device, "/dev/ttyE%c", funky_hex[i]);
+    snprintf(device, sizeof(device),"/dev/ttyE%c", funky_hex[i]);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
@@ -1040,7 +1053,7 @@ list_devices(void)
 
   for (i = 0; i < 128; i ++)
   {
-    sprintf(device, "/dev/ttyA%d", i + 1);
+    snprintf(device, sizeof(device), "/dev/ttyA%d", i + 1);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
